src/types.ts

```diff
@@ -1,20 +1,52 @@
-import { IComputedValue, IReactionOptions } from 'mobx';
+import { IReactionOptions } from 'mobx';
 import { StorageAdapter } from './StorageAdapter';
 
-export type PersistenceStore<T> = T & { _asJS: IComputedValue<string>; _storageName: string; _isPersistence: boolean };
-
-export type PersistenceDecoratorOptions = {
+export type PersistenceOptions = {
   name: string;
   properties: string[];
   adapter: StorageAdapter;
   reactionOptions?: IReactionOptions;
 };
 
-export type PersistenceCreatorReturnFunction = <T extends { new (...args: any): {} } | Object>(
-  target: T,
-) => PersistenceStore<T>;
-
 export type StorageAdapterOptions = {
```

Comment:
@quarrant I added some things to `StorageAdapterOptions`. Let me know your thoughts.